src/transaction.js

```diff
@@ -4,12 +4,15 @@ const EthObject = require('./ethObject')
 class Transaction extends EthObject{
 
   static get fields(){ return [
+    'chainId',
     'nonce',
-    'gasPrice',
+    'maxPriorityFeePerGas',
+    'maxFeePerGas',
     'gasLimit',
     'to',
     'value',
     'data',
+    'accessList',
     'v',
     'r',
     's',
```

Comment:
you'll need to have some conditionals here for backwards compatibility.

Reply:
Shouldn't it only encode fields that are set in the object and skips over the ones that aren't? I was wondering about this too but didn't look too deep into it when I saw the proofs were correct I assumed the fields for legacy txs are simply omited and has no effect on the encoding.